scipy/conftest.py

```diff
@@ -39,6 +39,30 @@ def pytest_runtest_setup(item):
     if mark is not None and np.intp(0).itemsize < 8:
         pytest.xfail('Fails on our 32-bit test platform(s): %s' % (mark.args[0],))
 
+    try:
+        from threadpoolctl import threadpool_limits
+        HAS_THREADPOOLCTL = True
+    except ImportError:
```

Comment:
`threadpoolctl` doesn't work well with `-OO`, see  joblib/threadpoolctl#87. In the mean time,

```suggestion
    except ImportError, AttributeError:
```

Reply:
Thanks for fixing things upstream Peter!